icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -2127,10 +2108,9 @@ void SimpleDateFormat::adoptNumberFormat(NumberFormat *formatToAdopt) {
         fSharedNumberFormatters = NULL;
     }
 
-    // Also re-compute the fast formatters.
-    UErrorCode localStatus = U_ZERO_ERROR;
-    freeFastNumberFormatters();
-    initFastNumberFormatters(localStatus);
+    // Null out the fast formatter, it's invalid now
+    delete fSimpleNumberFormatter;
+    fSimpleNumberFormatter = nullptr;
```

Comment:
*If* the fSimpleNumberFormatter shared DecimalFormatSymbols with fNumberFormatter, then you need to clean up fSimpleNumberFormatter first.